plugins/EventGhost/__init__.py

```diff
@@ -595,55 +595,292 @@ def __call__(self):
+        eventStringCtrl = panel.TextCtrl(eventString)
         waitTimeCtrl = panel.SpinNumCtrl(waitTime, integerWidth=5)
+        payloadCtrl = panel.TextCtrl(payload)
+        queueEventCtrl = wx.CheckBox(panel, -1, '')
+        restoreEventCtrl = wx.CheckBox(panel, -1, '')
+
+        queueEventCtrl.SetValue(queueEvent)
+        restoreEventCtrl.SetValue(restoreEvent)
+        queueEventCtrl.Enable(not waitTime)
+        restoreEventCtrl.Enable(not waitTime and not queueEvent)
+
+        def on_spin(evt):
+            def check_spin():
+                value = bool(waitTimeCtrl.GetValue())
+                queueEventCtrl.Enable(not value)
+                restoreEventCtrl.Enable(
+                    not value or (not value and not queueEventCtrl.GetValue())
+                )
+            wx.CallLater(20, check_spin)
+            evt.Skip()
+
+        def on_check(evt):
+            restoreEventCtrl.Enable(not queueEventCtrl.GetValue())
+            evt.Skip()
+
+        def HBoxSizer(lbl, ctrl, suf=None, prop=0):
+            sizer = wx.BoxSizer(wx.HORIZONTAL)
+            style = wx.EXPAND | wx.ALL | wx.ALIGN_BOTTOM
+
+            lbl = panel.StaticText(lbl)
+            lbl_sizer = wx.BoxSizer(wx.VERTICAL)
+            lbl_sizer.AddStretchSpacer(prop=1)
+            lbl_sizer.Add(lbl)
+
+            sizer.Add(lbl_sizer, 0, style, 5)
+            sizer.Add(ctrl, prop, style, 5)
+
+            if suf is not None:
+                suf = panel.StaticText(suf)
+                suf_sizer = wx.BoxSizer(wx.VERTICAL)
+                suf_sizer.AddStretchSpacer(prop=1)
+                suf_sizer.Add(suf)
+                sizer.Add(suf_sizer, 0, style, 5)
+            panel.sizer.Add(sizer, 0, wx.EXPAND)
+
+            return lbl
+
+        waitTimeCtrl.Bind(wx.EVT_SPIN, on_spin)
+        waitTimeCtrl.Bind(wx.EVT_CHAR_HOOK, on_spin)
+        queueEventCtrl.Bind(wx.EVT_CHECKBOX, on_check)
+
+        eg.EqualizeWidths((
+            HBoxSizer(text.text1, eventStringCtrl, prop=1),
+            HBoxSizer(text.text6, payloadCtrl, prop=1),
+            HBoxSizer(text.text2, waitTimeCtrl, suf=text.text3),
+            HBoxSizer(text.text4, queueEventCtrl),
+            HBoxSizer(text.text5, restoreEventCtrl)
+        ))
 
-        sizer1 = eg.HBoxSizer(
-            (panel.StaticText(text.text1), 0, wx.ALIGN_CENTER_VERTICAL, 5),
-            (eventStringCtrl, 0, wx.LEFT, 5),
-        )
-        sizer2 = eg.HBoxSizer(
-            (panel.StaticText(text.text2), 0, wx.ALIGN_CENTER_VERTICAL),
-            (waitTimeCtrl, 0, wx.ALL, 5),
-            (panel.StaticText(text.text3), 0, wx.ALIGN_CENTER_VERTICAL),
-        )
-        panel.sizer.AddMany(((sizer1, 0, wx.EXPAND), (sizer2, 0, wx.EXPAND)))
         while panel.Affirmed():
             panel.SetResult(
                 eventStringCtrl.GetValue(),
                 waitTimeCtrl.GetValue(),
+                payloadCtrl.GetValue() if payloadCtrl.GetValue() else None,
```

Comment:
replace `None` with `''` else you get an exception next time you wan't to edit the action.

Reply:
Payload has to be sent in as None if there isn't one. this is because it is the default value for TriggerEvent. So it makes it easier in the __call__ method if it can simply pass the None. so what I did was I just added a check for the None and to change it to ""